src/modules/job-list/job_data.c

```diff
@@ -123,22 +123,33 @@ static int parse_per_resource (struct job *job,
                                int *count)
 {
     json_error_t error;
+    json_t *o = NULL;
 
     if (json_unpack_ex (job->jobspec, &error, 0,
-                        "{s:{s:{s?:{s?:{s?:{s?:s s?:i}}}}}}",
+                        "{s:{s:{s?:{s?:{s?:o}}}}}",
                         "attributes",
                           "system",
                             "shell",
                               "options",
-                                "per-resource",
-                                  "type", type,
-                                  "count",count) < 0) {
+                                "per-resource", &o) < 0) {
         flux_log (job->h, LOG_ERR,
                   "%s: job %ju invalid jobspec: %s",
                   __FUNCTION__, (uintmax_t)job->id, error.text);
         return -1;
     }
 
+    if (o) {
+        if (json_unpack_ex (o, &error, 0,
```

Comment:
Looks like the job shell initializes `count` to `1` when the `per-resources` key exists, but `count` is not set.  If I'm reading the code correctly, this code would leave `count` set to `0`.

Reply:
ahhh i didn't realize that subtlety.  So it's more like count is optionally specified, assumed `1` is not specified.

Reply:
Yeah, sorry. In retrospect the current behavior seems like a bad idea, but sadly it is what it is.